### ISSUE TITLE
Build error on Mac OS X 10.10.5

### ISSUE THREAD

**Issue by celogeek** [net-amqp-rabbitmq/net-amqp-rabbitmq#57]
Hi,

I have build error on Mac OS X. I guess it is not handle properly ?

```
cpanm -v Net::AMQP::RabbitMQ
cpanm (App::cpanminus) 1.7014 on perl 5.012002 built for darwin-2level
Work directory is /Users/weborama/.cpanm/work/1441985951.83623
You have make /usr/bin/make
You have LWP 6.04
You have /usr/bin/tar: bsdtar 2.8.3 - libarchive 2.8.3
You have /usr/bin/unzip
Searching Net::AMQP::RabbitMQ on cpanmetadb ...
--> Working on Net::AMQP::RabbitMQ
Fetching http://www.cpan.org/authors/id/M/MA/MARKELLIS/Net-AMQP-RabbitMQ-0.900000.tar.gz ... OK
Unpacking Net-AMQP-RabbitMQ-0.900000.tar.gz
x Net-AMQP-RabbitMQ-0.900000/
x Net-AMQP-RabbitMQ-0.900000/amqp_socket.c
x Net-AMQP-RabbitMQ-0.900000/amqp_tcp_socket.c
x Net-AMQP-RabbitMQ-0.900000/RabbitMQ.xs
x Net-AMQP-RabbitMQ-0.900000/perl_math_int64.c
x Net-AMQP-RabbitMQ-0.900000/t/
x Net-AMQP-RabbitMQ-0.900000/t/013_headers.t
x Net-AMQP-RabbitMQ-0.900000/t/025_declare_exchange_with_ae.t
x Net-AMQP-RabbitMQ-0.900000/t/012_timeout.t
x Net-AMQP-RabbitMQ-0.900000/t/008_queue_declare.t
x Net-AMQP-RabbitMQ-0.900000/t/000_use_ok.t
x Net-AMQP-RabbitMQ-0.900000/t/011_hearbeat.t
x Net-AMQP-RabbitMQ-0.900000/t/001_declare_exchange.t
x Net-AMQP-RabbitMQ-0.900000/t/018_is_connected.t
x Net-AMQP-RabbitMQ-0.900000/t/022_array_headers.t
x Net-AMQP-RabbitMQ-0.900000/t/024_boolean_header_fields.t
x Net-AMQP-RabbitMQ-0.900000/t/009_no_route.t
x Net-AMQP-RabbitMQ-0.900000/t/015_declare_queue_with_headers.t
x Net-AMQP-RabbitMQ-0.900000/t/003_consume.t
x Net-AMQP-RabbitMQ-0.900000/t/021_json_utf8_in_and_out.t
x Net-AMQP-RabbitMQ-0.900000/t/002_publish.t
x Net-AMQP-RabbitMQ-0.900000/t/014_bind_with_headers.t
x Net-AMQP-RabbitMQ-0.900000/t/007_get.t
x Net-AMQP-RabbitMQ-0.900000/t/004_selfconsume.t
x Net-AMQP-RabbitMQ-0.900000/t/016_reject.t
x Net-AMQP-RabbitMQ-0.900000/t/023_numeric_header_fields.t
x Net-AMQP-RabbitMQ-0.900000/t/019_utf8_safety.t
x Net-AMQP-RabbitMQ-0.900000/t/005_noack.t
x Net-AMQP-RabbitMQ-0.900000/t/006_txn.t
x Net-AMQP-RabbitMQ-0.900000/t/020_dead_channel_close.t
x Net-AMQP-RabbitMQ-0.900000/t/010_qos.t
x Net-AMQP-RabbitMQ-0.900000/amqp_private.h
x Net-AMQP-RabbitMQ-0.900000/amqp_connection.c
x Net-AMQP-RabbitMQ-0.900000/Changes
x Net-AMQP-RabbitMQ-0.900000/amqp.h
x Net-AMQP-RabbitMQ-0.900000/amqp_framing.h
x Net-AMQP-RabbitMQ-0.900000/LICENSE-MIT
x Net-AMQP-RabbitMQ-0.900000/amqp_timer.c
x Net-AMQP-RabbitMQ-0.900000/amqp_url.c
x Net-AMQP-RabbitMQ-0.900000/LICENSE
x Net-AMQP-RabbitMQ-0.900000/amqp_consumer.c
x Net-AMQP-RabbitMQ-0.900000/amqp_framing.c
x Net-AMQP-RabbitMQ-0.900000/amqp_tcp_socket.h
x Net-AMQP-RabbitMQ-0.900000/RabbitMQ.pm
x Net-AMQP-RabbitMQ-0.900000/manual_tests/
x Net-AMQP-RabbitMQ-0.900000/manual_tests/001_dead_letter_headers.t
x Net-AMQP-RabbitMQ-0.900000/amqp_mem.c
x Net-AMQP-RabbitMQ-0.900000/amqp_timer.h
x Net-AMQP-RabbitMQ-0.900000/amqp_api.c
x Net-AMQP-RabbitMQ-0.900000/amqp_ssl_socket.h
x Net-AMQP-RabbitMQ-0.900000/META.json
x Net-AMQP-RabbitMQ-0.900000/amqp_table.c
x Net-AMQP-RabbitMQ-0.900000/MANIFEST
x Net-AMQP-RabbitMQ-0.900000/amqp_socket.h
x Net-AMQP-RabbitMQ-0.900000/xt/
x Net-AMQP-RabbitMQ-0.900000/xt/017_changes.t
x Net-AMQP-RabbitMQ-0.900000/xt/100_transaction_memory_leak.t
x Net-AMQP-RabbitMQ-0.900000/META.yml
x Net-AMQP-RabbitMQ-0.900000/ppport.h
x Net-AMQP-RabbitMQ-0.900000/perl_math_int64.h
x Net-AMQP-RabbitMQ-0.900000/typemap
x Net-AMQP-RabbitMQ-0.900000/Makefile.PL
Entering Net-AMQP-RabbitMQ-0.900000
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.04)
Running Makefile.PL
Configuring Net-AMQP-RabbitMQ-0.900000 ... Checking if your kit is complete...
Looks good
Warning (mostly harmless): No library found for -lrt
Generating a Unix-style Makefile
Writing Makefile for Net::AMQP::RabbitMQ
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.04)
Checking if you have Math::Int64 0.34 ... Yes (0.53)
Building and testing Net-AMQP-RabbitMQ-0.900000 ... cp RabbitMQ.pm blib/lib/Net/AMQP/RabbitMQ.pm
Running Mkbootstrap for Net::AMQP::RabbitMQ ()
chmod 644 "RabbitMQ.bs"
"/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/xsubpp"  -typemap "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/typemap" -typemap "typemap"  RabbitMQ.xs > RabbitMQ.xsc && mv RabbitMQ.xsc RabbitMQ.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000 -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' RabbitMQ.c
In file included from RabbitMQ.xs:9:
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:325:1: error: expected ')'
DECLARE_XTOXLL(hton)
^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
htonll
^
/usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:325:1: note: to match this '('
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
htonll
^
/usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
In file included from RabbitMQ.xs:9:
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:325:1: error: definition of
      builtin function '__builtin_constant_p'
DECLARE_XTOXLL(hton)
^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
htonll
^
/usr/include/sys/_endian.h:141:25: note: expanded from macro 'htonll'
#define htonll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
In file included from RabbitMQ.xs:9:
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:326:1: error: expected ')'
DECLARE_XTOXLL(ntoh)
^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
ntohll
^
/usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:30: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
                             ^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:326:1: note: to match this '('
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
ntohll
^
/usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:5: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
    ^
In file included from RabbitMQ.xs:9:
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:326:1: error: definition of
      builtin function '__builtin_constant_p'
DECLARE_XTOXLL(ntoh)
^
/Users/weborama/.cpanm/work/1441985951.83623/Net-AMQP-RabbitMQ-0.900000/amqp_private.h:291:26: note: expanded from macro
      'DECLARE_XTOXLL'
  static inline uint64_t func##ll(uint64_t val)     \
                         ^
<scratch space>:36:1: note: expanded from here
ntohll
^
/usr/include/sys/_endian.h:140:25: note: expanded from macro 'ntohll'
#define ntohll(x)       __DARWIN_OSSwapInt64(x)
                        ^
/usr/include/libkern/_OSByteOrder.h:78:6: note: expanded from macro '__DARWIN_OSSwapInt64'
    (__builtin_constant_p(x) ? __DARWIN_OSSwapConstInt64(x) : _OSSwapInt64(x))
     ^
4 errors generated.
make: *** [RabbitMQ.o] Error 1
FAIL
! Installing Net::AMQP::RabbitMQ failed. See /Users/weborama/.cpanm/work/1441985951.83623/build.log for details. Retry with --force to force install it.
```

**Comment by markwellis** [comment]
sorry i have no idea, i don't have a mac to test on.

that said, i think it's because we're using the the librabbitmq source files directly, and not building librabbitmq via it's configure script. 

infact, there's an issue there that says they've seen this problem, https://github.com/alanxz/rabbitmq-c/issues/206

and the fix is to compile with `-DHAVE_HTONLL`

i can't test this for you, but you should be able to make it work using somethign like

cpanm --look Net::AMQP::RabbitMQ
edit Makefile.PL
change the line `DEFINE => "-DAMQ_PLATFORM='\"Perl_$^O_$^V\"'",` to `DEFINE => "-DAMQ_PLATFORM='\"Perl_$^O_$^V\"' -DHAVE_HTONLL", `
cpanm --verbose .

if that works, can you let me know and i'll figure out a way of releasing the fix without breaking it for everyone else!

**Comment by celogeek** [comment]
Oh oh ! it works !

look :

```
~/Downloads/net-amqp-rabbitmq {master} $ perl Makefile.PL 
Warning (mostly harmless): No library found for -lrt
Using IV backend
Checking for stdlib.h... yes
Checking for stdarg.h... yes
Checking for string.h... yes
Checking for float.h... yes
Checking for assert.h... yes
Checking for ctype.h... yes
Checking for errno.h... yes
Checking for limits.h... yes
Checking for locale.h... yes
Checking for math.h... yes
Checking for setjmp.h... yes
Checking for signal.h... yes
Checking for stddef.h... yes
Checking for stdio.h... yes
Checking for time.h... yes
Checking for sys/types.h... yes
Checking for sys/stat.h... yes
Checking for memory.h... yes
Checking for strings.h... yes
Checking for inttypes.h... yes
Checking for stdint.h... yes
Checking for unistd.h... yes
Checking for int64_t... yes
Generating a Unix-style Makefile
Writing Makefile for Math::Int64
Writing MYMETA.yml and MYMETA.json
Generating a Unix-style Makefile
Writing Makefile for Net::AMQP::RabbitMQ
Writing MYMETA.yml and MYMETA.json
~/Downloads/net-amqp-rabbitmq {master} $ make
cp RabbitMQ.pm blib/lib/Net/AMQP/RabbitMQ.pm
cp lib/Math/Int64.pm ../blib/lib/Math/Int64.pm
cp lib/Math/UInt64.pm ../blib/lib/Math/UInt64.pm
cp lib/Math/Int64/die_on_overflow.pm ../blib/lib/Math/Int64/die_on_overflow.pm
cp lib/Math/Int64/native_if_available.pm ../blib/lib/Math/Int64/native_if_available.pm
Running Mkbootstrap for Math::Int64 ()
chmod 644 "Int64.bs"
"/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/xsubpp"  -typemap "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/typemap" -typemap "typemap"  Int64.xs > Int64.xsc && mv Int64.xsc Int64.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -O3   -DVERSION=\"0.49_01\" -DXS_VERSION=\"0.49_01\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DINT64_BACKEND_IV -DINT64_T Int64.c
rm -f ../blib/arch/auto/Math/Int64/Int64.bundle
env MACOSX_DEPLOYMENT_TARGET=10.3 cc  -bundle -undefined dynamic_lookup -L/usr/local/lib -fstack-protector Int64.o  -o ../blib/arch/auto/Math/Int64/Int64.bundle 	\
	     	\
	  
chmod 755 ../blib/arch/auto/Math/Int64/Int64.bundle
"/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Int64.bs ../blib/arch/auto/Math/Int64/Int64.bs 644
Manifying 4 pod documents
Running Mkbootstrap for Net::AMQP::RabbitMQ ()
chmod 644 "RabbitMQ.bs"
"/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/xsubpp"  -typemap "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/ExtUtils/typemap" -typemap "typemap"  RabbitMQ.xs > RabbitMQ.xsc && mv RabbitMQ.xsc RabbitMQ.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL RabbitMQ.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_api.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_connection.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_consumer.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_debug.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_framing.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_mem.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_socket.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_table.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_tcp_socket.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_timer.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_url.c
cc -c   -fno-common -DPERL_DARWIN -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -I/Users/weborama/Downloads/net-amqp-rabbitmq -O3   -DVERSION=\"0.900000\" -DXS_VERSION=\"0.900000\"  "-I/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/lib/5.12.2/darwin-2level/CORE"  -DAMQ_PLATFORM='"Perl_darwin_v5.12.2"' -DHAVE_HTONLL /Users/weborama/Downloads/net-amqp-rabbitmq/perl_math_int64.c
rm -f blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bundle
env MACOSX_DEPLOYMENT_TARGET=10.3 cc  -bundle -undefined dynamic_lookup -L/usr/local/lib -fstack-protector RabbitMQ.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_api.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_connection.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_consumer.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_debug.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_framing.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_mem.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_socket.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_table.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_tcp_socket.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_timer.o /Users/weborama/Downloads/net-amqp-rabbitmq/amqp_url.o /Users/weborama/Downloads/net-amqp-rabbitmq/perl_math_int64.o   -o blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bundle 	\
	     	\
	  
chmod 755 blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bundle
"/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- RabbitMQ.bs blib/arch/auto/Net/AMQP/RabbitMQ/RabbitMQ.bs 644
Manifying 1 pod document
~/Downloads/net-amqp-rabbitmq {master} $ make test
Running Mkbootstrap for Math::Int64 ()
chmod 644 "Int64.bs"
Manifying 4 pod documents
Running Mkbootstrap for Net::AMQP::RabbitMQ ()
chmod 644 "RabbitMQ.bs"
PERL_DL_NONLAZY=1 "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/000_use_ok.t ...................... ok   
t/001_declare_exchange.t ............ ok   
t/002_publish.t ..................... ok   
t/003_consume.t ..................... ok    
t/004_selfconsume.t ................. ok     
t/005_noack.t ....................... ok     
t/006_txn.t ......................... ok     
t/007_get.t ......................... ok     
t/008_queue_declare.t ............... ok   
t/009_no_route.t .................... ok   
t/010_qos.t ......................... ok   
t/011_hearbeat.t .................... 3/6 # Sleeping for 5 seconds
t/011_hearbeat.t .................... 5/6 # Sleeping for 1 seconds
t/011_hearbeat.t .................... ok   
t/012_timeout.t ..................... ok   
t/013_headers.t ..................... ok     
t/014_bind_with_headers.t ........... 4/17 # Using queue amq.gen-_1RY-vQodiqpcR9-5PNbBg
t/014_bind_with_headers.t ........... ok     
t/015_declare_queue_with_headers.t .. ok   
t/016_reject.t ...................... ok     
t/018_is_connected.t ................ ok   
t/019_utf8_safety.t ................. ok     
t/020_dead_channel_close.t .......... ok   
t/021_json_utf8_in_and_out.t ........ ok     
t/022_array_headers.t ............... ok     
t/023_numeric_header_fields.t ....... ok     
t/024_boolean_header_fields.t ....... ok     
t/025_declare_exchange_with_ae.t .... ok     
All tests successful.
Files=25, Tests=276, 46 wallclock secs ( 0.16 usr  0.07 sys +  0.80 cusr  0.18 csys =  1.21 CPU)
Result: PASS
PERL_DL_NONLAZY=1 "/Users/weborama/perl5/perlbrew/perls/perl-5.12.2/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, '../blib/lib', '../blib/arch')" t/*.t
t/as_int64.t ............ ok   
t/die_on_overflow.t ..... ok    
t/Math-Int64-Native.t ... ok    
t/Math-Int64.t .......... ok      
t/Math-UInt64-Native.t .. ok    
t/Math-UInt64.t ......... ok     
t/MSC.t ................. ok   
t/pow.t ................. ok    
t/storable.t ............ ok     
All tests successful.
Files=9, Tests=2524,  1 wallclock secs ( 0.19 usr  0.04 sys +  0.41 cusr  0.04 csys =  0.68 CPU)
Result: PASS
```

Can we have a fix with in on cpan ?

**Comment by markwellis** [comment]
excellent! I'll have a think of a way to detect when it is needed and get it on cpan.

**Comment by celogeek** [comment]
If you have a branch I can test I can confirm that your solution works.

**Comment by markwellis** [comment]
this is released in version 1.1, which will be on cpan shortly
